### PR TITLE
MGDOBR-1228: Adopt TableView from app-services-ui-components

### PR DESCRIPTION
**Pull request by vpellegrino**
**Task:** https://issues.redhat.com/browse/MGDOBR-1228

After these changes, we have almost all IT tests failing because they were based on `ouiaId`s or different DOM structures.

### CODE REVIEW

**Comment by vpellegrino** [comment]
@kelvah, apart from the test failures that we were already expecting, there is also one change in the UI's behavior.
When we had no instances/processors previously it was like this:
image
Now, with `TableView`, we should adopt a solution like the one used by the Kafka Instances table:
https://redhat-developer.github.io/app-services-ui-components/?path=/docs/kafka-kafkainstances-components-emptystatenoinstances--empty-state-no-instances

Is it ok for you? Should we also involve UX in that change, or is it expected with the adoption of this component?

**Comment by kelvah** [comment]
> @kelvah, apart from the test failures that we were already expecting, there is also one change in the UI's behavior. Is it ok for you? Should we also involve UX in that change, or is it expected with the adoption of this component?

@vpellegrino we chatted about this on our call earlier. Just to keep track of things also here, the new behavior was agreed with UX team already. So fine for me to follow their guidelines and switch to the "empty state" replacing the table.

**Comment by sjamboro** [comment]
When I have started to create Mocked and Dev Quality Checks I was followed the OUIA Standard which looked as the good practice:
https://www.patternfly.org/v4/developer-resources/open-ui-automation/
https://docs.cypress.io/guides/references/best-practices

The label might reflect the UX experience better but it might be a source of instability like change lables, find substrings, etc.

In case that there are a lot of changes and the UI offers new views, user's useages, adopt the `v2` backend. We should disable quality checks and consider new implementation. What do you think @Srihari1192 . 
The OUIA supports was added by qe members and sometimes it was not trivial. I understand that the implementation of ouia standard might slow down a progress on UI. The new component brings new behaviour (it requires more wait rutine).

**Comment by Srihari1192** [comment]
@sjamboro Yes agree, if our new UI is adopting to v2 SDK, we can disable the quality checks for now as it's failing.

File: src/app/Instance/InstancePage/ProcessorsTabContent.tsx [hunk 1]
```diff
@@ -217,37 +196,36 @@ export const ProcessorsTabContent = ({
...
+        <TableView
+          ariaLabel={t(
+            "smartEventsTempDictionary:processor.processorsListTable"
+          )}
+          actions={[
+            /** TODO https://issues.redhat.com/browse/MGDOBR-1274 */
+            {
+              label: t("processor.createProcessor"),
+              onClick: onCreateProcessor,
+              isPrimary: true,
```
Comment (sjamboro) [inline src/app/Instance/InstancePage/ProcessorsTabContent.tsx:208]: There is not possible to add ouiaId. The action should be the create button in toolbar.

File: src/app/Instance/InstancesListPage/InstancesListPage.tsx [hunk 2]
```diff
@@ -223,93 +213,82 @@ const InstancesListPage = (): JSX.Element => {
...
     ({ row, ActionsColumn }): JSX.Element => (
       <ActionsColumn items={tableActions(row as IRowData)} />
     ),
     [tableActions]
   );
+
+  const onCreateInstance = (): void => setShowCreateInstance(true);
+
+  const emptyStateNoData = (
+    <EmptyState
```
Comment (sjamboro) [inline src/app/Instance/InstancesListPage/InstancesListPage.tsx:232]: It is our implementation we can add ouiaId here easily, but the create button is visible only if there is no data (bridges / processes).

**Review by sjamboro — COMMENTED** [review]
There are my thought about this code.

**Comment by vpellegrino** [comment]
> There are my thought about this code. This [migration](https://issues.redhat.com/browse/MGDOBR-1178) hides ouiaId implementation and there is not support to set `data-ouia-component-id`. In such situation the only reasonable solution is `contains("button", "Create processor).`. I know that it is a common approach in UI. I was solving several issues when labels was substring of another one and it broke our test. Currently we have possibility to find it directly in PR so it could not be pain. This PR changes only a small piece of test suite but in case that we will adopt more components it could be a reason why we lost a benefits of ouia standards.

@sjamboro I am not sure I get all points you mentioned in the review you submitted.
BTW, I am 100% open to including in this PR different solution. So, feel free to open a PR against my branch.

File: cypress/integration/instance/InstancePage.ts [hunk 3]
```diff
@@ -148,9 +148,7 @@ onlyOn(isEnvironmentType(EnvType.Mocked), () => {
         .ouiaId("processors", "PF4/TabButton")
         .click();
       cy.ouiaId("processors", "PF4/TabContent").within(() => {
-        cy.ouiaId("rows-toolbar", "PF4/Toolbar")
-          .ouiaId("create-processor", "PF4/Button")
-          .should("be.visible");
+        cy.contains("button", "Create processor").should("be.visible");
```
[the remark this answers is not in the document]
Comment (vpellegrino) [reply cypress/integration/instance/InstancePage.ts:151]: We don't have a `rows-toolbar` anymore. It was part of the previous component, `TableWithPagination`.
The new one does not have any more that identifier.

File: cypress/integration/instance/InstancePage.ts [hunk 4]
```diff
@@ -188,9 +186,7 @@ onlyOn(isEnvironmentType(EnvType.Mocked), () => {
...
```
[the remark this answers is not in the document]
Comment (vpellegrino) [reply cypress/integration/instance/InstancePage.ts:189]: We don't have a `rows-toolbar` ouia-id anymore. It was part of the previous component, `TableWithPagination`.
The new one (from app-services-ui-component) does not have this id anymore.

File: src/app/components/EmptyState/EmptyState.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+                onClick={quickStartGuide.onQuickstartGuide}
+              />,
+            ]}
+          />
+        </EmptyStateBody>
+      )}
+      <Button
+        variant="primary"
+        onClick={createButton.onCreate}
+        isDisabled={createButton.isDisabled}
```
[the remark this answers is not in the document]
Comment (vpellegrino) [reply src/app/components/EmptyState/EmptyState.tsx:54]: We could add it, but in this version of the tests, we access the button by its label.

File: src/app/Instance/InstancePage/ProcessorsTabContent.tsx (hunk 1 above)
Comment (vpellegrino) [reply src/app/Instance/InstancePage/ProcessorsTabContent.tsx:208]: I confirm. It is not currently possible to pass an ouiaId property to the Action configuration of the table.

File: src/app/Instance/InstancesListPage/InstancesListPage.tsx (hunk 2 above)
Comment (vpellegrino) [reply src/app/Instance/InstancesListPage/InstancesListPage.tsx:232]: Exactly. This is the same observation I made in https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/pull/269#discussion_r1033967337

File: cypress/integration/instance/InstancePage.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (sjamboro) [reply cypress/integration/instance/InstancePage.ts:151]: @vpellegrino I checked that toolbar is still here but we are not able to assign the ouiaId. You have added "action" and the collector for these actions is "PF4/Toolbar". We can use `cy.ouiaType("PF4/Toolbar") ...`

File: cypress/integration/instance/InstancePage.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (sjamboro) [reply cypress/integration/instance/InstancePage.ts:189]: @vpellegrino I checked that toolbar is still here but we are not able to assign the ouiaId. You have added "action" and the collector for these actions is "PF4/Toolbar". We can use cy.ouiaType("PF4/Toolbar") ...

File: src/app/components/EmptyState/EmptyState.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (sjamboro) [reply src/app/components/EmptyState/EmptyState.tsx:54]: @vpellegrino yes I agree with you. This covered only the empty list. It does not make sense to continue in OUIA.

File: cypress/integration/instance/InstancePage.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (vpellegrino) [reply cypress/integration/instance/InstancePage.ts]: @sjamboro https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/pull/269/commits/0af9b9aff51d2362c33e8d38a7c6be58340528d8

File: cypress/integration/instance/InstancePage.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (vpellegrino) [reply cypress/integration/instance/InstancePage.ts]: @sjamboro https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/pull/269/commits/0af9b9aff51d2362c33e8d38a7c6be58340528d8

File: src/app/components/EmptyState/EmptyState.tsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  title: string;
+}
+
+export const EmptyState = ({
+  createButton,
+  quickStartGuide,
+  title,
+}: EmptyStateProps): JSX.Element => {
+  return (
+    <EmptyStatePF variant={EmptyStateVariant.xs}>
```
Comment (kelvah) [inline src/app/components/EmptyState/EmptyState.tsx:32]: Using the `small` variant will add a bit of extra padding around the empty state. It looks too close to the parent now.

```suggestion
    <EmptyStatePF variant={EmptyStateVariant.small}>
```

**Review by kelvah — APPROVED** [review]
LGTM :+1: 

I really like that the pagination params are now persisted into the page URL.
Nice work!

**Comment by vpellegrino** [comment]
@sjamboro, can you please review this PR and let me know if you have further considerations?

**Review by sjamboro — APPROVED** [review]
Thank you @vpellegrino. The IT tests are green. They should cover:
- create button is available and works
- pagination is available and works
- bridge instance are showed correctly

**Pull request merged**